Update create-config-file.md (#4262)
grafana/agent: docs/sources/static/configuration/create-config-file.md

```

In this example, we first must configure the `wal_directory` which is used to

> store metrics in a Write-Ahead Log. This is not required, but ensures that samples

Reviewer: ```suggestion
store metrics in a Write-Ahead Log (WAL). WAL is required and ensures that samples
```

> will be redelivered in case of failure (e.g., network issues, machine reboot). We

Reviewer: ```suggestion
will be redelivered in case of failures, for example if there are network connectivity issues. We
```

@mattdurham would this suggestion cover the point you raised about reboots/restarts not handled by WAL?

Reviewer 2: Yes, that is primary use case.
